projects/angular/src/data/datagrid/datagrid-filter.ts

```diff
@@ -90,7 +91,8 @@ export class ClrDatagridFilter<T = any>
     _filters: FiltersProvider<T>,
     public commonStrings: ClrCommonStringsService,
     private smartToggleService: ClrPopoverToggleService,
-    @Inject(PLATFORM_ID) private platformId: any
+    @Inject(PLATFORM_ID) private platformId: any,
+    private elementRef: ElementRef<HTMLElement>
```

Comment:
Can't we make this optional, so we don't change the API too much?

Reply:
Hmm, logically it does not make sense. I was looking at the test when I wrote the above comment, but it's highly unlikely that anyone will use the constructor directly outside of testing scenarios. Please disregard the previous comment.

Reply:
Users should definitely not be calling directive/component constructors. I consider these parameters not part of the public API.